packages/react-graphql/src/hooks/graphql-document.ts

```diff
@@ -52,10 +53,11 @@ export default function useGraphQLDocument<
     [document, loadDocument],
   );
 
-  return [
-    document,
+  useAsyncAsset(
```

Comment:
I also moved this back into this hook, where it feels more appropriate than the `useQuery` hook